src/main/java/org/folio/rest/impl/BLUsersAPI.java

```diff
@@ -1152,64 +1158,62 @@ private Future<List<String>> getLocateUserFields(List<String> fieldAliasList, Ma
    * @return
    */
   private Future<User> locateUserByAlias(List<String> fieldAliasList, Identifier entity,
-                                                       Map<String, String> okapiHeaders, String errorKey) {
-    Promise<User> asyncResult = Promise.promise();
-    getLocateUserFields(fieldAliasList, okapiHeaders).onComplete(locateUserFieldsAR -> {
-      if (!locateUserFieldsAR.succeeded()) {
-        asyncResult.fail(locateUserFieldsAR.cause());
-        return;
-      }
-      String query = buildQuery(locateUserFieldsAR.result(), entity.getId());
-
-      String tenant = okapiHeaders.get(OKAPI_TENANT_HEADER);
-      String okapiURL = okapiHeaders.get(OKAPI_URL_HEADER);
-
-      HttpClientInterface client = HttpClientFactory.getHttpClient(okapiURL, tenant);
-
-      okapiHeaders.remove(OKAPI_URL_HEADER);
+                                         Map<String, String> okapiHeaders, String errorKey) {
+    return getLocateUserFields(fieldAliasList, okapiHeaders)
+      .compose(locateUserFieldsAR -> consortiaService.locateConsortiaUser(entity.getId(), okapiHeaders, errorKey)
+        .compose(user -> {
+          if (user == null) {
+            return locateUser(locateUserFieldsAR, entity, okapiHeaders, errorKey);
+          }
+          return Future.succeededFuture(user);
+          }));
+  }
 
-      try {
-        String userUrl = new StringBuilder("/users?").append("query=").append(URLEncoder.encode(query, "UTF-8")).append("&").append("offset=0&limit=2").toString();
+  private Future<User> locateUser(List<String> locateUserFieldsAR, Identifier entity, Map<String, String> okapiHeaders, String errorKey) {
+    String tenant = okapiHeaders.get(OKAPI_TENANT_HEADER);
+    String okapiURL = okapiHeaders.get(OKAPI_URL_HEADER);
+    Promise<User> asyncResult = Promise.promise();
+    HttpClientInterface client = HttpClientFactory.getHttpClient(okapiURL, tenant);
+    String query = buildQuery(locateUserFieldsAR, entity.getId());
+    try {
+      String userUrl = new StringBuilder("/users?").append("query=").append(URLEncoder.encode(query, "UTF-8")).append("&").append("offset=0&limit=2").toString();
```

Comment:
I think a better approach here is using String.format OR UriBuilder 
```
String encodedQuery = URLEncoder.encode(query, StandardCharsets.UTF_8);
String userUrl = String.format("/users?query=%s&offset=0&limit=2", encodedQuery);
```

```
UriBuilder builder = UriBuilder.fromPath("/users")
    .queryParam("query", URLEncoder.encode(query, StandardCharsets.UTF_8))
    .queryParam("offset", "0")
    .queryParam("limit", "2");
URI userUrl = builder.build();
```

Reply:
refactored